feature_engine/selection/recursive_feature_addition.py

```diff
@@ -0,0 +1,269 @@
+from typing import List, Union
+
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import cross_validate
+from sklearn.utils.validation import check_is_fitted
+
+from feature_engine.dataframe_checks import (
+    _is_dataframe,
+    _check_input_matches_training_df,
+)
+from feature_engine.selection.base_selector import get_feature_importances
+from feature_engine.variable_manipulation import (
+    _check_input_parameter_variables,
+    _find_or_check_numerical_variables,
+)
+
+Variables = Union[None, int, str, List[Union[str, int]]]
+
+
+class RecursiveFeatureAddition(BaseEstimator, TransformerMixin):
+    """
+
+    RecursiveFeatureAddition selects features. #WIP
+
+
+    Model training and performance calculation are done with cross-validation.
+
+    Parameters
+    ----------
+
+    variables : str or list, default=None
+        The list of variable to be evaluated. If None, the transformer will evaluate
+        all numerical features in the dataset.
+
+    estimator: object, default = RandomForestClassifier()
+        A Scikit-learn estimator for regression or classification.
+        The estimator must have either a feature_importances or coef attribute
+        after fitting.
+
+    scoring: str, default='roc_auc'
+        Desired metric to optimise the performance of the estimator. Comes from
+        sklearn.metrics. See the model evaluation documentation for more options:
+        https://scikit-learn.org/stable/modules/model_evaluation.html
+
+    threshold: float, int, default = 0.01
+        The value that defines if a feature will be kept or removed. Note that for
+        metrics like roc-auc, r2_score and accuracy, the thresholds will be floats
+        between 0 and 1. For metrics like the mean_square_error and the
+        root_mean_square_error the threshold will be a big number.
+        The threshold must be defined by the user. Bigger thresholds will retain less
+        features.
+
+    cv : int, default=3
+        Cross-validation fold to be used to fit the estimator.
+
+    Attributes
+    ----------
+
+    initial_model_performance_: float
+        performance of the model built using the original dataset with all the features.
+
+    feature_importances_: pandas series
+        A pandas Series containing the feature names in the axis, and the performance
+        derived from the model trained on the entire dataset, as values. The Series is
+        ordered from least important to most important feature.
+
+    performance_drifts_: dict
+        A dictionary containing the feature, and the change in performance incurred
+        when training a model without that feature.
+
+
+    Methods
+    -------
+
+    fit: finds important features
+
```

Comment:
Supposedly, we do not need to add a description here to the methods. It should be picked it automatically by numpydoc from the fit method dosctring below. To this end, when we describe fit and transform, we should add a short initial sentence, which is the one that will be populated in this section of the documentation. 

I am unifying and revamping the docs now, you can see the latest format in any transformer (except the selection ones) in this PR:
https://github.com/solegalli/feature_engine/pull/184